crates/core/src/requests/mod.rs

```diff
@@ -0,0 +1,1224 @@
+
+use serde::de::Error as _;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use world_id_primitives::rp::{RpId, RpNullifierKey};
+use world_id_primitives::{FieldElement, PrimitiveError, WorldIdProof};
+
+/// Protocol schema version for proof requests and responses.
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum RequestVersion {
+    /// Version 1
+    V1 = 1,
+}
+
+impl serde::Serialize for RequestVersion {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let v = *self as u8;
+        serializer.serialize_u8(v)
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for RequestVersion {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let v = u8::deserialize(deserializer)?;
+        match v {
+            1 => Ok(Self::V1),
+            _ => Err(serde::de::Error::custom("unsupported version")),
+        }
+    }
+}
+
+/// A proof request from a relying party for an authenticator
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct ProofRequest {
+    /// Unique identifier for this request
+    pub id: String,
+    /// Version of the request
+    pub version: RequestVersion,
+    /// Unix timestamp (seconds since epoch) when the request was created
+    pub created_at: u64,
+    /// Unix timestamp (seconds since epoch) when request expires
+    pub expires_at: u64,
+    /// Registered RP id
+    pub rp_id: RpId,
+    /// The raw representation of the action. This must be already a field element.
+    ///
+    /// When dealing with strings or bytes, such value can be hashed e.g. with a byte-friendly
+    /// hash function like keccak256 or SHA256 and then reduced to a field element.
+    pub action: FieldElement,
+    /// The nullifier key of the RP
+    pub rp_nullifier_key: RpNullifierKey,
+    /// The RP's ECDSA signature over the request
+    pub signature: k256::ecdsa::Signature,
+    /// Unique nonce for this request (serialized as hex string)
+    pub nonce: FieldElement,
+    /// Specific credential requests. This defines which credentials to ask for.
+    #[serde(rename = "proof_requests")]
+    pub requests: Vec<RequestItem>,
+    /// Constraint expression (all/any) optional
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub constraints: Option<ConstraintExpr<'static>>,
+}
+
+/// Per-credential request payload
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct RequestItem {
+    /// The specific credential being requested as registered in the `CredentialIssuerSchemaRegistry`.
+    /// Serialized as hex string in JSON.
+    pub issuer_schema_id: FieldElement,
+    /// Optional signal (TODO: improve documentation)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub signal: Option<String>,
```

Comment:
here we want to add other fields, like `face_auth` boolean, which we added here https://github.com/worldcoin/idkit/blob/main/js/packages/core/src/lib/hashing.ts#L276

Reply:
I don't think we should add additional attributes at the protocol level until those are supported by the protocol.